### PR TITLE
DAOS-2938: test: Handle agent startup failure

### PR DESCRIPTION
**Pull request by brianjmurrell**

### CODE REVIEW

File: src/tests/ftest/util/agent_utils.py [hunk 1]
```diff
@@ -151,11 +148,21 @@ def run_agent(basepath, server_list, client_list=None):
...
 
             match = re.findall(pattern, output)
-            if len(match) > 0:
+            if match:
+                started_clients.append(client)
                 print("<AGENT> agent started on node {} in {} "
                       "seconds".format(client, time.time() - start_time))
                 break
 
+        return_code = sessions[client].poll()
```
Comment (phender) [inline src/tests/ftest/util/agent_utils.py]: Given that the previous `.poll()` call in the `while` loop would have already set the return code, would it be slightly more efficient to use this instead:
```suggestion
        return_code = sessions[client].returncode
```

File: src/tests/ftest/util/agent_utils.py [hunk 2]
```diff
@@ -151,11 +148,21 @@ def run_agent(basepath, server_list, client_list=None):
...
+        if return_code is not None:
+            print("Uh-oh, in agent startup, the ssh that started the agent "
```
Comment (phender) [inline src/tests/ftest/util/agent_utils.py]: Should we maintain the "<AGENT> ..." message format used in the other print statements?

File: src/tests/ftest/util/agent_utils.py [hunk 3]
```diff
@@ -114,28 +115,24 @@ def run_agent(basepath, server_list, client_list=None):
...
-            client,
-            daos_agent_bin,
-            "-i"
-        ]
-
-        p = subprocess.Popen(cmd,
-                             stdout=subprocess.PIPE,
-                             stderr=subprocess.PIPE)
-        sessions[client] = p
+        cmd = ["ssh", client, "{} -i".format(daos_agent_bin)]
```
Comment (phender) [inline src/tests/ftest/util/agent_utils.py]: Completely optional - in #793 I was able to remove the too-many-locals disabled linting error with this:

```
sessions[client] = subprocess.Popen(
    ["ssh", client, "{} -i".format(daos_agent_bin)],
    stdout=subprocess.PIPE,
    stderr=subprocess.PIPE)
```

File: src/tests/ftest/util/agent_utils.py [hunk 4]
```diff
@@ -27,23 +27,24 @@
...
 import socket
 import errno
 import fcntl
 import re
 
 class AgentFailed(Exception):
-    pass
+    """ _ """
 
+# pylint: disable=too-few-public-methods
```
[the remark this answers is not in the document]
Comment (brianjmurrell) [reply src/tests/ftest/util/agent_utils.py]: Done

File: src/tests/ftest/util/agent_utils.py (hunk 1 above)
Comment (brianjmurrell) [reply src/tests/ftest/util/agent_utils.py]: Done

File: src/tests/ftest/util/agent_utils.py (hunk 2 above)
Comment (brianjmurrell) [reply src/tests/ftest/util/agent_utils.py]: Done

File: src/tests/ftest/util/agent_utils.py (hunk 3 above)
Comment (brianjmurrell) [reply src/tests/ftest/util/agent_utils.py]: Actually, that _too-many-locals disabled linting_ was prior to my editing of this file and I must have already eliminated enough locals as I was able to remove those without the suggested change.  But I don't dislike the suggestion so implemented it anyway.  :-)

**Pull request merged**